### PR TITLE
Download the correct CLI Template files

### LINKED ISSUE #673: Download the correct CLI template files
### Describe the bug

- The `cli-template-contracts-hardhat` is not downloading the `.yarn` folder and the `yarnrc.yml` file.

- The `cli-template-monorepo-ethers` and `cli-template-monorepo-hardhat` templates are downloading folders that are not necessary like: `typechain-types`, `artifacts` and `cache`.

### PR DESCRIPTION
**Pull request by vplasencia**
## Related Issue(s)

Closes #673

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have run `yarn prettier` and `yarn lint` without getting any errors
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/cli-template-contracts-hardhat/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
         "test:report-gas": "REPORT_GAS=true hardhat test",
         "test:coverage": "hardhat coverage",
         "typechain": "hardhat typechain",
-        "prepublish": "tar -czf files.tgz .gitignore",
+        "prepublish": "tar --exclude typechain-types --exclude artifacts --exclude cache  --exclude node_modules -czf files.tgz .gitignore .yarn .yarnrc.yml",
```
Comment (cedoor) [inline packages/cli-template-contracts-hardhat/package.json]: I'm not sure if we need to explicitly exclude those folders. I've just tried to run `tar -czf files.tgz .gitignore` and even if there are those folders (typechain-types, artifacts, node_modules), they've not been included.

Comment (vplasencia) [reply packages/cli-template-contracts-hardhat/package.json]: If you run `semaphore create` and select the `monorepo-ethers` template for example, you will see the `artifacts` and `typechain-types` there inside `apps/contracts`. Then we can include those files to make sure they will never be there or make sure that the files are not there when releasing a new version of the packages. 

I think that with automatic releases, we wouldn't need to exclude the files.

Comment (cedoor) [reply packages/cli-template-contracts-hardhat/package.json]: Sounds right for the monorepos, but we shouldn't need to exclude them in the contracts-hardhat template right?

Comment (vplasencia) [reply packages/cli-template-contracts-hardhat/package.json]: Yes, you're right. That's not necessary for this specific template, just for the others. I will update it.

File: packages/cli-template-contracts-hardhat/package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
         "test:report-gas": "REPORT_GAS=true hardhat test",
```
Comment (cedoor) [inline packages/cli-template-contracts-hardhat/package.json:30]: I think `.yarn` and `.yarnrc.yml` can also be removed from the files list.

**Review by cedoor — CHANGES_REQUESTED** [review]

File: packages/cli-template-contracts-hardhat/package.json (hunk 2 above)
Comment (vplasencia) [reply packages/cli-template-contracts-hardhat/package.json:30]: The project is using `yarn` and if these files are not downloaded, devs would need to at least create a `.yarnrc.yml` file. We could remove them when we adapt the templates to use other package manager by default like `npm` for example. What do you think?

For this specific template we could remove the `.yarn` folder and modify the `yarnrc.yml` file to have only `nodeLinker: node-modules`.

Comment (cedoor) [reply packages/cli-template-contracts-hardhat/package.json:30]: Sorry, I meant from the package.json's `files` field list.

**Review by cedoor — APPROVED** [review]

**Pull request merged**